mintpy/utils/isce_utils.py

```diff
@@ -917,3 +917,123 @@ def get_sensing_datetime_list(proj_dir, date_list=None):
 
     return sensingMid, sensingStart, sensingStop
 
+############################## Standard Processing ###########################################
+
+def gaussian_kernel(Sx, Sy, sig_x, sig_y):
+    '''
+    Generate a guassian kernal (with all elements sum to 1)
+    inputs: Sx, Sy - dimensions of kernal
+            sig_x, sig_y: standard deviation of the guassian distribution
+    '''
+    if np.mod(Sx,2) == 0:
+        Sx = Sx + 1
+
+    if np.mod(Sy,2) ==0:
+            Sy = Sy + 1
+
+    x,y = np.meshgrid(np.arange(Sx),np.arange(Sy))
+    x = x + 1
+    y = y + 1
+    x0 = (Sx+1)/2
+    y0 = (Sy+1)/2
+    fx = ((x-x0)**2.)/(2.*sig_x**2.)
+    fy = ((y-y0)**2.)/(2.*sig_y**2.)
+    k = np.exp(-1.0*(fx+fy))
+    a = 1./np.sum(k)
+    k = a*k
+    return k
+
+def convolve(data, kernel):
+    '''
+    return a convolved (filtered) complex image
+    inputs: data - complex array
+            kernel - convolution kernel
+    '''
+    from scipy import ndimage
+    R = ndimage.convolve(data.real, kernel, mode='constant',cval=0.0)
+    Im =ndimage.convolve(data.imag, kernel, mode='constant',cval=0.0)
+
+    return R + 1J*Im
+
+def estimate_coherence(infile, corfile):
+    '''
+    input: infile : .int file path
+            corfile: output correlation file (computed from phase sigma)
+    '''
+    import isce
+    import isceobj
+    from mroipac.icu.Icu import Icu
+
+    #Create phase sigma correlation file here
+    filtImage = isceobj.createIntImage()
+    filtImage.load( infile + '.xml')
+    filtImage.setAccessMode('read')
+    filtImage.createImage()
+
+    phsigImage = isceobj.createImage()
+    phsigImage.dataType='FLOAT'
+    phsigImage.bands = 1
+    phsigImage.setWidth(filtImage.getWidth())
+    phsigImage.setFilename(corfile)
+    phsigImage.setAccessMode('write')
+    phsigImage.createImage()
+
+
+    icuObj = Icu(name='sentinel_filter_icu')
+    icuObj.configure()
+    icuObj.unwrappingFlag = False
+    icuObj.useAmplitudeFlag = False
+    #icuObj.correlationType = 'NOSLOPE'
+
+    icuObj.icu(intImage = filtImage,  phsigImage=phsigImage)
+    phsigImage.renderHdr()
+
+    filtImage.finalizeImage()
+    phsigImage.finalizeImage()
+
+    return
+
+def unwrap_snaphu(intfile,corfile,unwfile, meta ,cost='SMOOTH'):
+    '''
+    input: intfile - wrapped phase file directory
+           corfile - correlation file directory
+           unwfile - output unwrapped file directory
+           meta - dict, attributes dictionary
+           cost - 'DEFO,'SMOOTH','TOPO'
+    '''
+    import isce
+    from contrib.Snaphu.Snaphu import Snaphu
+
+    width = int(meta['width'])
+    wavelength = float(meta['WAVELENGTH'])
+    altitude =  float(meta['altitude'])
```

Comment:
```suggestion
    altitude =  float(meta['HEIGHT'])
```

`HEIGHT` would be a more generic name than `altitude` in mintpy, as the latter is copied from `isce2` and might not be there if the input dataset is using a different insar processor.

Reply:
Thanks for pointing it out. Replaced as suggested.